include/swift/SIL/SILInstruction.h

```diff
@@ -7855,60 +7855,69 @@ class DifferentiableFunctionInst final :
 private:
   friend SILBuilder;
   /// Differentiation parameter indices.
-  AutoDiffIndexSubset *parameterIndices;
-  /// The order of differentiation.
-  unsigned differentiationOrder;
-  /// The number of operands. The first operand is always the original function.
-  /// The rest of operands determined by the order of differentiation and whether
-  /// this is the new AD model or the legacy reverse-mode AD model.
-  unsigned numOperands;
+  AutoDiffIndexSubset *ParameterIndices;
+  /// Indicates whether derivative functions (JVP/VJP) exist.
+  bool HasDerivativeFunctions;
 
-  DifferentiableFunctionInst(SILModule &module, SILDebugLocation debugLoc,
-                             AutoDiffIndexSubset *parameterIndices,
-                             unsigned differentiationOrder,
-                             SILValue originalFunction,
-                             ArrayRef<SILValue> associatedFunctions);
+  DifferentiableFunctionInst(
+      SILDebugLocation DebugLoc, AutoDiffIndexSubset *ParameterIndices,
+      SILValue OriginalFunction, ArrayRef<SILValue> DerivativeFunctions,
+      bool HasOwnership);
+
+  static SILType getAutoDiffType(
```

Comment:
Minor unrelated renaming suggestion:
```suggestion
  static SILType getDifferentiableFunctionType(
```

(the utility computes a `@differentiable` version of the original function type of the given `SILValue`)

Reply:
Done.